build error when `LINK_STATIC_LIBPROTOBUF=ON` and `make -j` are used

```
$ cmake -DLINK_STATIC_LIBPROTOBUF=ON .. && make -j4
...
Scanning dependencies of target gen_onnx_proto
Scanning dependencies of target Protobuf
[  1%] Running gen_proto.py on onnx/onnx.in.proto
[  2%] Creating directories for 'Protobuf'
[  3%] Performing download step (download, verify and extract) for 'Protobuf'
-- Downloading...
   dst='/Users/sakai/Menoh/build/downloads/protobuf-cpp-3.6.1.tar.gz'
   timeout='none'
-- Using src='https://github.com/protocolbuffers/protobuf/releases/download/v3.6.1/protobuf-cpp-3.6.1.tar.gz'
Processing /Users/sakai/Menoh/external/onnx/onnx/onnx.in.proto
Writing /Users/sakai/Menoh/build/external/onnx/onnx/onnx.proto
Writing /Users/sakai/Menoh/build/external/onnx/onnx/onnx.proto3
generating /Users/sakai/Menoh/build/external/onnx/onnx/onnx_pb.py
[  4%] Running C++ protocol buffer compiler on /Users/sakai/Menoh/build/external/onnx/onnx/onnx.proto
/bin/sh: ../../protobuf-3.6.1/bin/protoc: No such file or directory
make[2]: *** [external/onnx/onnx/onnx.pb.cc] Error 127
make[1]: *** [external/onnx/CMakeFiles/gen_onnx_proto.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
-- [download 0% complete]
...
-- [download 100% complete]
-- verifying file...
       file='/Users/sakai/Menoh/build/downloads/protobuf-cpp-3.6.1.tar.gz'
-- Downloading... done
-- extracting...
     src='/Users/sakai/Menoh/build/downloads/protobuf-cpp-3.6.1.tar.gz'
     dst='/Users/sakai/Menoh/build/protobuf-3.6.1/src/Protobuf'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[  6%] No patch step for 'Protobuf'
[  7%] No update step for 'Protobuf'
[  8%] Performing configure step for 'Protobuf'
...
```

[full log](https://github.com/pfnet-research/menoh/files/2476508/log.txt)

--- a/cmake/BuildProtobuf.cmake
+++ b/cmake/BuildProtobuf.cmake
@@ -1,8 +1,8 @@
 set(PROTOBUF_VERSION_STATIC "3.6.1")
+set(PROTOBUF_HASH MD5=406d5b8636576b1c86730ca5cbd1e576)
 
 set(PROTOBUF_DIR ${CMAKE_CURRENT_BINARY_DIR}/protobuf-${PROTOBUF_VERSION_STATIC})
 set(PROTOBUF_URL "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION_STATIC}/protobuf-cpp-${PROTOBUF_VERSION_STATIC}.tar.gz")
-set(PROTOBUF_HASH MD5=406d5b8636576b1c86730ca5cbd1e576)
 
 # Requires `-fPIC` for linking with a shared library
 set(PROTOBUF_CFLAGS "-g -O2 -fPIC")
@@ -34,11 +34,16 @@ set(PROTOBUF_LIBRARIES ${PROTOBUF_LIBRARY})
 set(PROTOBUF_PROTOC_EXECUTABLE ${PROTOBUF_DIR}/bin/protoc)
 set(PROTOBUF_FOUND TRUE)
 
+# configure protobuf::libprotobuf
 add_library(protobuf::libprotobuf UNKNOWN IMPORTED)
 # Note: INTERFACE_INCLUDE_DIRECTORIES can't set in this place because include/ is
 # not installed during executing `cmake`
 set_target_properties(protobuf::libprotobuf PROPERTIES
     IMPORTED_LOCATION "${PROTOBUF_LIBRARY_STATIC}")
+add_dependencies(protobuf::libprotobuf Protobuf)
+
+# configure protobuf::protoc
 add_executable(protobuf::protoc IMPORTED)
 set_target_properties(protobuf::protoc PROPERTIES
     IMPORTED_LOCATION "${PROTOBUF_PROTOC_EXECUTABLE}")
+add_dependencies(protobuf::protoc Protobuf)

--- a/scripts/build-menoh.sh
+++ b/scripts/build-menoh.sh
@@ -63,8 +63,8 @@ test -n "${ARG_BUILD_DIR}" || readonly ARG_BUILD_DIR="${ARG_SOURCE_DIR}/build"
 test -n "${ARG_INSTALL_DIR}" || readonly ARG_INSTALL_DIR=/usr/local
 
 if [ -n "${ARG_MKLDNN_DIR}" ]; then
-    OPT_MKLDNN_INCLUDE_DIR=-DMKLDNN_INCLUDE_DIR=${ARG_MKLDNN_DIR}/include
-    OPT_MKLDNN_LIBRARY=-DMKLDNN_LIBRARY=${ARG_MKLDNN_DIR}/lib/libmkldnn.so
+    readonly OPT_MKLDNN_INCLUDE_DIR=-DMKLDNN_INCLUDE_DIR=${ARG_MKLDNN_DIR}/include
+    readonly OPT_MKLDNN_LIBRARY=-DMKLDNN_LIBRARY=${ARG_MKLDNN_DIR}/lib/libmkldnn.so
 fi
 
 test -n "${ARG_PYTHON_EXECUTABLE}" || readonly ARG_PYTHON_EXECUTABLE=python

